src/app/app_ui.rs

```diff
@@ -0,0 +1,388 @@
+enum GetImageError {
+    AddressNotValid,
+    MemoryReadError(std::io::Error),
+}
+
+impl App for Application {
+    fn update(&mut self, ctx: &Context, _frame: &mut eframe::Frame) {
+        CentralPanel::default().show(ctx, |ui| {
+            if self.in_settings {
+                self.draw_settings(ui);
+                if ui.input(|i| i.key_pressed(Key::Escape)) {
+                    self.in_settings = false;
+                }
+            } else {
+                ui.horizontal(|ui| {
+                    ui.vertical(|ui| self.draw_config(ui));
+
+                    ui.separator();
+
+                    ui.vertical(|ui| self.draw_image_block(ui));
+                });
+            }
+        });
+
+        TopBottomPanel::bottom("bottom_panel").show(ctx, |ui| {
+            ui.horizontal(|ui| {
+                self.logger.borrow().show_label(ui);
+                ui.with_layout(Layout::right_to_left(Align::Max), |ui| {
+                    if ui.button("⛭").clicked() {
+                        self.in_settings = !self.in_settings;
+                    }
+                });
+            });
+        });
+    }
+
+    fn save(&mut self, storage: &mut dyn eframe::Storage) {
+        eframe::set_value(storage, eframe::APP_KEY, self);
+    }
+}
+
+impl Application {
+    fn draw_config(&mut self, ui: &mut eframe::egui::Ui) {
+        ui.set_width(200f32);
+        // Process selection
+        let mut text_edit_output = None;
+        if ui
+            .menu_button(self.config.pid_label.clone(), |ui| {
+                self.sysinfo.refresh();
+                text_edit_output = Some(
+                    TextEdit::singleline(&mut self.sysinfo.search_filter)
+                        .hint_text("Search process")
+                        .show(ui),
+                );
+                self.draw_processes(ui);
+            })
+            .response
+            .clicked()
+        {
+            if let Some(mut teo) = text_edit_output.take() {
+                let cc_range = CCursorRange::two(
+                    CCursor::new(0),
+                    CCursor::new(self.sysinfo.search_filter.len()),
+                );
+                teo.state.cursor.set_char_range(Some(cc_range));
+                teo.state.store(ui.ctx(), teo.response.id);
+                teo.response.request_focus();
+            }
+        }
+
+        // Memory address input
+        ui.horizontal(|ui| {
+            if !self.config.address.is_empty() && parse_address(&self.config.address).is_err() {
+                ui.style_mut().visuals.override_text_color =
+                    Some(ui.style().visuals.error_fg_color);
+            }
+            ui.add(TextEdit::singleline(&mut self.config.address).hint_text("Memory address"));
+            let reread_shortcut = KeyboardShortcut::new(Modifiers::COMMAND, Key::R);
+            let reread_shortcut_pressed = ui.input_mut(|i| i.consume_shortcut(&reread_shortcut));
+
+            if ui
+                .button("↺")
```

Comment:
Large if conditions are harder to read.
It's better to extract a variable to hold the expression result